akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -657,10 +658,26 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
           (remoteGossip, !remoteGossip.seenByNode(selfUniqueAddress), Newer)
         case _ ⇒
           // conflicting versions, merge
-          (remoteGossip merge localGossip, true, Merge)
+          val prunedLocalGossip = localGossip.members.foldLeft(localGossip) { (g, m) ⇒
```

Comment:
> Scenario 1: M is contained in both gossips. In this case it will stay in both pruned versions, and as a result in the merged version

Right, then there has not been any removal (by the leader) and no pruning is involved.

> Scenario 2: M is contained in only one gossip (lets name it G1, and G2 is the other gossip)

G1_pruned will still contain the member M. Pruning only clears entry in vclock. Removal of M itself is handled by `merge` (`pickHighestPriority`).

The idea is (if I remember correctly) that this merge logic for the conflicting case will do the same pruning as the leader did. We see that it was removed because it is not in one of the gossips, and we know that it existed before and is on the way out since the other has status Down or Exiting.